binding/web/src/koala.ts

```diff
@@ -251,20 +293,45 @@ export class Koala {
     modelPath: string,
     options: KoalaOptions = {}
   ): Promise<Koala> {
+    let { device = "best" } = options;
     const { processErrorCallback } = options;
 
     if (!isAccessKeyValid(accessKey)) {
       throw new KoalaErrors.KoalaInvalidArgumentError('Invalid AccessKey');
     }
 
+    const isSimd = await simd();
+    if (!isSimd) {
+      throw new KoalaErrors.KoalaRuntimeError('Browser not supported.');
+    }
+
+    if (device === "best") {
+      device = "cpu:1";
+    }
+
+    const isWorkerScope = typeof WorkerGlobalScope !== 'undefined' && self instanceof WorkerGlobalScope;
+    if (!isWorkerScope) {
+      if (device && CPU_DEVICE_REGEX.test(device)) {
+        if (device !== "cpu" && device !== "cpu:1") {
+          console.warn("Multi-threading is not supported on main thread.");
+        }
+        device = "cpu:1";
+      }
+    }
+
+    const sabDefined = (typeof SharedArrayBuffer !== 'undefined')
+      && (device !== "cpu") && (device !== "cpu:1");
```

Comment:
this is confusing - why do we want to change is if `cpu` or `cpu:1` are defined? Did you just mean `cpu:1`?